pkg/scheduler/framework/plugins/migration/utils.go

```diff
@@ -47,3 +47,44 @@ func ErrorToFrameworkStatus(err error) *framework.Status {
 	}
 	return nil
 }
+
+const (
+	// PredicatesStateKey is the key in CycleState to PredicateStateData
+	PredicatesStateKey = "predicates"
+
+	// PrioritiesStateKey is the key in CycleState to PrioritiesStateData
+	PrioritiesStateKey = "priorities"
+)
+
+// PredicatesStateData is a pointer to PredicateMetadata. In the normal case, StateData is supposed to
+// be generated and stored in CycleState by a framework plugin (like a PreFilter pre-computing data for
+// its corresponding Filter). However, during migration, the scheduler will inject a pointer to
+// PredicateMetadata into CycleState. This "hack" is necessary because during migration Filters that implement
+// predicates functionality will be calling into the existing predicate functions, and need
+// to pass PredicateMetadata.
+type PredicatesStateData struct {
+	Reference interface{}
```

Comment:
In the future, when we finish the migration, we can move the fields of `PredicateMetadata` here? (right now it holds a pointer pointing to metaCopy)

Reply:
This is holding a pointer to the PredicatesMetadata that is currently used, not that we copy the CycleState every time we copy meta, and so they are one-to-one all the time.

We need this during phase 1 of the migration phase where we cal the predicates/priorities directly and so we need the metadata. In phase 2, this will be removed, metadata producers will be moved to be prefilters, each will write its pre-computed data (e.g., evenpodsspread prefilter will generate and write evenpodsspread pre-computed data), see this PR: https://github.com/kubernetes/kubernetes/pull/83286

Reply:
SG. Thanks for explanation.